src/models/inbound/dfspConsentRequests.model.ts

```diff
@@ -0,0 +1,281 @@
+        { name: 'storeReqAndSendOTP', from: 'RequestIsValid', to: 'success' }
+      ],
+      methods: {
+        // specific transitions handlers methods
+        onValidateRequest: () => this.onValidateRequest(),
+        onStoreReqAndSendOTP: () => this.onStoreReqAndSendOTP()
+      }
+    }
+    super(data, config, spec)
+    this.config = { ...config }
+  }
+
+  // getters
+  get pubSub (): PubSub {
+    return this.config.pubSub
+  }
+
+  get dfspBackendRequests (): DFSPBackendRequests {
+    return this.config.dfspBackendRequests
+  }
+
+  get thirdpartyRequests (): ThirdpartyRequests {
+    return this.config.thirdpartyRequests
+  }
+
+  async onValidateRequest (): Promise<void> {
+    const { request, toParticipantId } = this.data
+
+    try {
+      const response = await this.dfspBackendRequests.validateConsentRequests(request)
+
+      if (!response) {
+        throw new Errors.MojaloopFSPIOPError(
+          null as unknown as string,
+          null as unknown as string,
+          null as unknown as string,
+          Errors.MojaloopApiErrorCodes.TP_FSP_CONSENT_SCOPES_ERROR
+        )
+      }
+
+      if (!response.isValid) {
+        throw new Errors.MojaloopFSPIOPError(
+          null as unknown as string,
+          null as unknown as string,
+          null as unknown as string,
+          Errors.MojaloopApiErrorCodes.TP_NO_SUPPORTED_SCOPE_ACTIONS
+        )
+      }
+
+      this.data.response = response
+      if (response.authChannels.includes('WEB')) {
+        const webConsentRequestResponse: tpAPI.Schemas.ConsentRequestsIDPutResponseWeb = {
+          scopes: request.scopes,
+          callbackUri: request.callbackUri,
+          authChannels: ['WEB'],
+          authUri: response.authUri!,
+          initiatorId: toParticipantId
+        }
+        await this.thirdpartyRequests.putConsentRequests(request.id, webConsentRequestResponse, toParticipantId)
+      } else {
+        const otpConsentRequestResponse: tpAPI.Schemas.ConsentRequestsIDPutResponseOTP = {
+          scopes: request.scopes,
+          callbackUri: request.callbackUri,
+          authChannels: ['OTP'],
+          initiatorId: toParticipantId
+        }
+        await this.thirdpartyRequests.putConsentRequests(request.id, otpConsentRequestResponse, toParticipantId)
+      }
+
+    } catch (error) {
+      const mojaloopError = this.reformatError(error)
+
+      this.logger.push({ error }).error('start -> RequestIsValid')
+      this.logger.push({ mojaloopError }).info(`Sending error response to ${toParticipantId}`)
+
+      await this.thirdpartyRequests.putConsentRequestsError(
+        request.id,
+        mojaloopError as unknown as fspiopAPI.Schemas.ErrorInformationObject,
+        toParticipantId
+      )
+
+      // throw error to stop state machine
+      throw error
+    }
+  }
+
+  async onStoreReqAndSendOTP (): Promise<void> {
+    const { request, response } = this.data
+
+    try {
+      if (response?.authChannels.includes('WEB')) {
```

Comment:
Can you please explain this to me? Is there a case where `authChannels` has 2 elements? Or do we validate for that elsewhere?

I would prefer something more readable and future proof like:

```ts
const channel = response.authChannels.pop();
switch (channel): {
  case 'WEB': ..
  case 'OTP':...
  default: ...
}
```

Reply:
when PISP sends  `POST /consentRequests` to DFSP, it validates that request scopes and responds with 
`PUT /consentRequests`or `PUT /consentRequests/{id}/error`
response contains  2 extra fields . 1 authChannels['WEB' or 'OTP']  2. authUri(only in case  of web)
these details returned from dfsp back-end or mojaloop-simulator 
here if else checks response authChannels[] and if it's web, returns authChannels, authUri otherwise only authChannels details to pisp
as there are only 2 scenarios I used if -else,  will  change it to switch  
`Is there a case where authChannels has 2 elements?  I am not sure about this, it's design decision`

Reply:
Ok now I have modified the code to forward back-end response as it is without those checks

Reply:
I'm still seeing the `if else` block..

Reply:
can you pls refresh,  I see new changes
https://github.com/mojaloop/thirdparty-scheme-adapter/blob/4cac6a7f00b9e7a5e9e102fbfb0013dcd1038a6c/src/models/inbound/dfspConsentRequests.model.ts